scripts/g.extension/g.extension.py

```diff
@@ -270,25 +271,29 @@ def get_github_branches(
 
 def get_default_branch(full_url):
     """Get default branch for repo (currently only implemented for github API and gitlab API)"""
-    if "/github.com/" in full_url:
-        organization, repository = re.split(r"/github.com/", full_url)[1].split("/")[
-            0:2
+    # Parse URL
+    url_parts = urlparse(full_url)
+    # Get organization and repository component
+    try:
+        organization, repository = url_parts.path.split("/")[
+            1:3
         ]
+    except URLError:
+        gscript.fatal(_("Cannot retrieve organization and repository from URL: <{}>.".format(full_url)))
```

Comment:
Having this part here (and not e.g. in a separate, possibly nested function and called only when needed), prohibits general URLs to be used here (just to get `main`). I'm not sure how relevant that is, though.

Reply:
This function is only meant for known hosting services (currently: github, gitlab, bitbucket), so, this should not be an issue...

Reply:
I was thinking the `main` would be used as a the default for any other service, but in that case, use likely needs to (and can) specify branch explicitly anyway.